### PR TITLE
Sprint5

### PR DESCRIPTION
**Pull request by Lehren**
Sprint 5 endpoint

### CODE REVIEW

File: src/main/java/org/fsg1/fmms/backend/database/Connection.java [hunk 1]
```diff
@@ -67,18 +68,25 @@ public String executeQuery(final String columnName, final String statement, fina
...
-     * @return A possible result in JSON string form if columnName is given, else an empty string is returned.
+     * @return The generated INSERT id if an INSERT was made, or 0 if no INSERT was made but, for instance,
+     * a DELETE or UPDATE.
      * @throws SQLException If a database access error occurs or anything else goes wrong.
      */
-    public int executeUpdate(final java.sql.Connection connection,
-                             final String statement,
-                             final Object... parameters) throws SQLException {
-        try (PreparedStatement preparedStatement = connection.prepareStatement(statement)) {
+    public long executeUpdate(final java.sql.Connection connection,
```
Comment (lukeelten) [inline src/main/java/org/fsg1/fmms/backend/database/Connection.java]: Can be int. Database only store 4-byte integer as primary keys

File: src/main/java/org/fsg1/fmms/backend/database/Connection.java [hunk 2]
```diff
@@ -67,18 +68,25 @@ public String executeQuery(final String columnName, final String statement, fina
...
-            return preparedStatement.executeUpdate();
-        } catch (Exception e) {
-            connection.rollback();
-            throw e;
+            preparedStatement.executeUpdate();
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    return generatedKeys.getLong(1);
+                }
+                return 0;
```
Comment (lukeelten) [inline src/main/java/org/fsg1/fmms/backend/database/Connection.java]: This should not happen, maybe error handling?

Comment (Lehren) [reply src/main/java/org/fsg1/fmms/backend/database/Connection.java]: It shouldn't? What will it return on a delete, or update?

Comment (lukeelten) [reply src/main/java/org/fsg1/fmms/backend/database/Connection.java]: Return -1 and check everytime you depend on this result whether it is real or not

File: src/main/java/org/fsg1/fmms/backend/endpoints/EditableModuleEndpoint.java [hunk 3]
```diff
@@ -90,6 +93,50 @@ public Response postModuleInformation(@PathParam("module_id") final int moduleId
...
+                        dependency.findValue("id").asInt(),
+                        dependency.findValue("type").asText(),
+                        dependency.findValue("remarks").asText());
+            }
+
+            service.update(conn, queries[10],
+                    id);
+
+            for (JsonNode learningGoal : learningGoals) {
+                int generatedId = service.update(conn, queries[11],
```
Comment (lukeelten) [inline src/main/java/org/fsg1/fmms/backend/endpoints/EditableModuleEndpoint.java:113]: Check if generated id is not an error code

**Review by lukeelten — CHANGES_REQUESTED** [review]

**Review by lukeelten — APPROVED** [review]

**Pull request merged**